pkg/engine/engine.go

```diff
@@ -1094,6 +1228,30 @@ func (e *Engine) detectChunk(ctx context.Context, data detectableChunk) {
 		}
 
 		for _, res := range results {
+			if e.enableDetectionCache {
+				// Cache the result to avoid re-verifying the same secret.
+				hashKeyBytes := make([]byte, len(res.Raw)+len(res.RawV2))
+				copy(hashKeyBytes, res.Raw)
+				copy(hashKeyBytes[len(res.Raw):], res.RawV2)
```

Comment:
Same question as above in `verificationOverlapWorker` (which should be removed IMO...)

Reply:
After thinking about this for a bit, I worry the implementation isn't intuitive and/or is hard to follow. I can't quite put my finger on it... the change itself isn't bad by any means.

I expected most of the logic to be in `detectChunk` but a lot of the heavy lifting is done in `verificationOverlapWorker` (in general). Those functions share a decent amount of duplicate code (e.g., why do we need to call `processResult` in both?), yet the relevant bits are close to 200 lines a part.

This is probably an issue greater than the scope of this PR.

Reply:
@ahrav Moreover, inserting `verificationOverlapWorker` has completely broken the `--allow-verification-overlap` flag:
```sh
$ ./trufflehog filesystem /tmp/overlap --allow-verification-overlap
🐷🔑🐷  TruffleHog. Unearth your secrets. 🐷🔑🐷

Found unverified result 🐷🔑❓
Verification issue: More than one detector has found this result. For your safety, verification has been disabled.You can override this behavior by using the --allow-verification-overlap flag.
...
```

Because of: 
```diff
-		if len(matchingDetectors) > 1 && !e.verificationOverlap {
				wgVerificationOverlap.Add(1)
				e.verificationOverlapChunksChan <- verificationOverlapChunk{
					chunk:                       *decoded.Chunk,
					detectors:                   matchingDetectors,
					decoder:                     decoded.DecoderType,
					verificationOverlapWgDoneFn: wgVerificationOverlap.Done,
				}
```